### PR TITLE
Add proj4_dict_to_str utility function

### PR DESCRIPTION
**Pull request by davidh-ssec**
Includes fixes for dynamic area definitions proj_id and
small performance improvement for projection coordinate generation

I know it isn't the best that I included some of these changes in one PR, but I had them sitting around and thought I needed to do it.

### CODE REVIEW

**Comment by davidh-ssec** [comment]
I should note that this PR should work with and without the changes by @goodsonr in #77.

File: pyresample/geometry.py [hunk 1]
```diff
@@ -1041,17 +1045,9 @@ def get_val(val, sub_val, max):
...
-                                    cols), dtype=dtype)
-
-        target_y = np.fromfunction(lambda i, j:
-                                   self.pixel_upper_left[1] -
-                                   (i + row_start) * self.pixel_size_y,
-                                   (rows,
-                                    cols), dtype=dtype)
+        target_x = np.arange(col_start, col_start + cols, dtype=dtype) * self.pixel_size_x + self.pixel_upper_left[0]
+        target_y = np.arange(row_start, row_start + rows, dtype=dtype) * -self.pixel_size_y + self.pixel_upper_left[1]
+        target_x, target_y = np.meshgrid(target_x, target_y)
```
Comment (mraspaud) [inline pyresample/geometry.py]: have you checked the difference in performance with meshgrid ?

File: pyresample/test/test_utils.py [hunk 2]
```diff
@@ -221,6 +221,16 @@ def test_proj4_radius_parameters_default(self):
         np.testing.assert_almost_equal(a, 6378137.)
         np.testing.assert_almost_equal(b, 6356752.314245, decimal=6)
 
+    def proj4_str_dict_conversion(self):
+        from pyresample import utils
+        proj_str = "+proj=lcc +ellps=WGS84 +lon_0=-95"
+        proj_dict = utils.proj4_str_to_dict(proj_str)
+        proj_str2 = utils.proj4_dict_to_str(proj_dict)
+        proj_dict2 = utils.proj4_str_to_dict(proj_str2)
+        for k, v in proj_dict2.items():
```
Comment (mraspaud) [inline pyresample/test/test_utils.py]: please avoid one letter variables in new code...

File: pyresample/utils.py [hunk 3]
```diff
@@ -410,6 +410,27 @@ def proj4_str_to_dict(proj4_str):
     return dict((x[0], (x[1] if len(x) == 2 else True)) for x in pairs)
 
 
+def proj4_dict_to_str(proj4_dict, sort=False):
+    """Convert a dictionary of PROJ.4 parameters to a valid PROJ.4 string"""
+    keys = proj4_dict.keys()
+    if sort:
+        keys = sorted(keys)
+    params = []
+    for k in keys:
```
Comment (mraspaud) [inline pyresample/utils.py]: please avoid one letter variables in new code...

File: pyresample/geometry.py (hunk 1 above)
Comment (davidh-ssec) [reply pyresample/geometry.py]:
```
In [1]: import numpy as np

In [2]: import timeit

In [4]: def test_old(rows, cols):
   ...:     dtype = np.float32
   ...:     col_start = 0
   ...:     row_start = 0
   ...:     pixel_size_x = 1000.
   ...:     pixel_size_y = 1000.
   ...:     pixel_upper_left = (-10000., 10000.)
   ...:     target_x = np.fromfunction(lambda i, j: (j + col_start) *
   ...:                                     pixel_size_x +
   ...:                                     pixel_upper_left[0],
   ...:                                     (rows,
   ...:                                      cols), dtype=dtype)
   ...:     target_y = np.fromfunction(lambda i, j:
   ...:                                     pixel_upper_left[1] -
   ...:                                     (i + row_start) * pixel_size_y,
   ...:                                     (rows,
   ...:                                      cols), dtype=dtype)
   ...:     return
   ...:                                     

In [5]: def test_new(rows, cols):
   ...:     dtype = np.float32
   ...:     col_start = 0
   ...:     row_start = 0
   ...:     pixel_size_x = 1000.
   ...:     pixel_size_y = 1000.
   ...:     pixel_upper_left = (-10000., 10000.)
   ...:     target_x = np.arange(col_start, col_start + cols, dtype=dtype) * pixel_size_x + pixel_upper_left[0]
   ...:     target_y = np.arange(row_start, row_start + rows, dtype=dtype) * -pixel_size_y + pixel_upper_left[1]
   ...:     target_x, target_y = np.meshgrid(target_x, target_y)
   ...:     return
   ...: 

In [9]: timeit.timeit('test_old(5000, 5000)', globals=globals(), number=100)
Out[9]: 35.2982305699843

In [10]: timeit.timeit('test_new(5000, 5000)', globals=globals(), number=100)
Out[10]: 6.261023567989469
```

**Pull request merged**